core/src/test/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackendSuite.scala

```diff
@@ -184,4 +193,47 @@ class CoarseMesosSchedulerBackendSuite extends SparkFunSuite
 
     verify(driver, times(1)).reviveOffers()
   }
+
+  test("isOfferSatisfiesRequirements return true when there is a valid offer") {
+    val schedulerBackend = createSchedulerBackendForGivenSparkConf(sc)
+
+    assert(schedulerBackend.isOfferSatisfiesRequirements("Slave1", 10000, 5, sc))
+  }
+
+
+  test("isOfferSatisfiesRequirements return false when memory in offer is less" +
+    " than required memory") {
+    val schedulerBackend = createSchedulerBackendForGivenSparkConf(sc)
+
+    assert(schedulerBackend.isOfferSatisfiesRequirements("Slave1", 1, 5, sc) === false)
+  }
+
+  test("isOfferSatisfiesRequirements return false when cpu in offer is less than required cpu") {
+    val schedulerBackend = createSchedulerBackendForGivenSparkConf(sc)
+
+    assert(schedulerBackend.isOfferSatisfiesRequirements("Slave1", 10000, 0, sc) === false)
+  }
+
+  test("isOfferSatisfiesRequirements return false when offer is from slave already running" +
+    " an executor") {
+    val schedulerBackend = createSchedulerBackendForGivenSparkConf(sc)
+    schedulerBackend.slaveIdsWithExecutors += "Slave2"
+
+    assert(schedulerBackend.isOfferSatisfiesRequirements("Slave2", 10000, 5, sc) === false)
+  }
+
+  test("isOfferSatisfiesRequirements return false when task is failed more than " +
+    "MAX_SLAVE_FAILURES times on the given slave") {
+    val schedulerBackend = createSchedulerBackendForGivenSparkConf(sc)
+    schedulerBackend.failuresBySlaveId("Slave3") = 2
+
+    assert(schedulerBackend.isOfferSatisfiesRequirements("Slave3", 10000, 5, sc) === false)
+  }
+
+  test("isOfferSatisfiesRequirements return false when max core is already acquired") {
```

Comment:
these tests can probably all be grouped in one test called `offerSatisfiesRequirements`.